### PR TITLE
libmdbx 0.9.2 (new formula)

### PR DESCRIPTION
**Pull request by ged**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Embeddable ACID-compliant key-value database library.

### CODE REVIEW

File: Formula/libmdbx.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    man.mkpath
+    man1.install(*MANPAGES)
+  end
+
+  test do
+    srcfile = testpath/"dbentries.txt"
+    dbdir = testpath/"test.mdbx"
+
+    srcfile.write("foo\nbar\nbaz\nbatim\n")
```
Comment (carlocab) [inline Formula/libmdbx.rb:28]:
```suggestion
    dbdir = testpath/"test.mdbx"

    (testpath/"dbentries.txt").write <<~EOS
      foo
      bar
      baz
      batim
    EOS

```

Comment (ged) [reply Formula/libmdbx.rb:28]: If it was intended to be meaningful, I'd agree, but it's not. It also breaks the test.

File: Formula/libmdbx.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    system "make", "all"
+
+    bin.install(*TOOLS)
+
+    lib.install "libmdbx.dylib"
+    lib.install "libmdbx.a"
+    include.install "mdbx.h", "mdbx.h++"
+
+    man.mkpath
+    man1.install(*MANPAGES)
```
Comment (SMillerDev) [inline Formula/libmdbx.rb:21]: Is there no `make install`?

File: Formula/libmdbx.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+class Libmdbx < Formula
+  desc "One of the fastest embeddable key-value ACID database without WAL"
+  homepage "https://erthink.github.io/libmdbx/"
+  url "https://github.com/erthink/libmdbx/releases/download/v0.9.2/libmdbx-amalgamated-0.9.2.tar.gz"
+  sha256 "c35cc53d66d74ebfc86e39441ba26276541ac7892bf91dba1e70c83665a02767"
+  license "OLDAP-2.8"
+
+  TOOLS = %w[mdbx_stat mdbx_copy mdbx_dump mdbx_load mdbx_chk].freeze
+  MANPAGES = TOOLS.map { |tool| "man1/#{tool}.1" }.freeze
```
Comment (SMillerDev) [inline Formula/libmdbx.rb:9]: These should just be inlined.

File: Formula/libmdbx.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  test do
+    srcfile = testpath/"dbentries.txt"
+    dbdir = testpath/"test.mdbx"
+
+    srcfile.write("foo\nbar\nbaz\nbatim\n")
+    system "#{bin}/mdbx_load", "-f", srcfile, "-T", dbdir
+
+    assert_predicate dbdir, :directory?
+    assert_match /No error is detected/i, shell_output("#{bin}/mdbx_chk #{dbdir}")
```
Comment (SMillerDev) [inline Formula/libmdbx.rb:32]:
```suggestion
    assert_match "No error is detected", shell_output("#{bin}/mdbx_chk #{dbdir}")
```

File: Formula/libmdbx.rb (hunk 2 above)
Comment (ged) [reply Formula/libmdbx.rb:21]: There is but it assumes flags for `install` that don't work on MacOS.

File: Formula/libmdbx.rb (hunk 3 above)
Comment (ged) [reply Formula/libmdbx.rb:9]: Okay.

File: Formula/libmdbx.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/libmdbx.rb]: Could you submit a patch upstream to make it work on macOS? That's much cleaner than us picking our own install.

Comment (ged) [reply Formula/libmdbx.rb]: I don't have time to do so, but maybe someone else can pick up this PR if/when that happens. Closing for now.

**Pull request closed without merging**